### ISSUE TITLE
IBM Plex Cond not named properly in CSS and filenames?

### ISSUE THREAD

**Issue by johnnyshields** [IBM/plex#316]
In the CSS for 5.0.0 I see:

```
@font-face {
  font-family: 'IBM Plex Sans Condensed';
  font-style: normal;
  font-weight: 200;
  src: local("IBM Plex Sans Condensed ExtraLight"), local("IBMPlexSansCondensed-ExtraLight"), url("../IBM-Plex-Sans-Condensed/fonts/split/woff2/IBMPlexSansCondensed-ExtraLight-Latin1.woff2") format("woff2");
```

Shouldn't this be `local("IBM Plex Sans Cond ExtraLight")`? Wouldn't it also make sense to change the filenames for Condensed to Cond?

**Comment by BoldMonday** [comment]
Good call. I think the naming for addressing local fonts should be changed indeed to the following:
```
Full Font Name                       PS Font Name
---------------------------------------------------------------
IBM Plex Sans Cond Bold Italic   |   IBMPlexSans-CondBoldItalic
IBM Plex Sans Cond Bold          |   IBMPlexSans-CondBold
IBM Plex Sans Cond ExtLt Italic  |   IBMPlexSans-CondExtLtItalic
IBM Plex Sans Cond ExtLt         |   IBMPlexSans-CondExtLt
IBM Plex Sans Cond Italic        |   IBMPlexSans-CondItalic
IBM Plex Sans Cond Light Italic  |   IBMPlexSans-CondLightItalic
IBM Plex Sans Cond Light         |   IBMPlexSans-CondLight
IBM Plex Sans Cond Medm Italic   |   IBMPlexSans-CondMedmItalic
IBM Plex Sans Cond Medm          |   IBMPlexSans-CondMedm
IBM Plex Sans Cond               |   IBMPlexSans-Cond
IBM Plex Sans Cond SmBld Italic  |   IBMPlexSans-CondSmBldItalic
IBM Plex Sans Cond SmBld         |   IBMPlexSans-CondSmBld
IBM Plex Sans Cond Text Italic   |   IBMPlexSans-CondTextItalic
IBM Plex Sans Cond Text          |   IBMPlexSans-CondText
IBM Plex Sans Cond Thin Italic   |   IBMPlexSans-CondThinItalic
IBM Plex Sans Cond Thin          |   IBMPlexSans-CondThin
```

**Comment by johnnyshields** [comment]
Another unrelated point. I find it a bit strange that the name "KR" is used for Korean while other scripts (Hebrew, Thai, etc.) are spelled in full. If we were to use ISO abbreviations, `KR` is actually for the country South Korea while `ko` is the Korean language.

**Comment by BoldMonday** [comment]
This has been discussed at length with the Korean designers at Sandoll, and the current naming has been their suggestion. Essentially it follows Noto.

**Comment by BoldMonday** [comment]
@mjabbink I think this should be assigned to @vpicone. We didn't create the css files in question.

**Comment by vpicone** [comment]
@johnnyshields using the ISO abbreviation was discussed, but left in favor of the current solution.

I'll take a look at the generated css when I can, should be a simple fix if you'd like to try a PR!

**Comment by BoldMonday** [comment]
@vpicone I'd like to help where I can but have no idea where to look. Do you use an algorithm to generate the fonts names for the `local` parameter?

**Comment by vpicone** [comment]
@BoldMonday I'll have a fix for this up soon, I'm going to change the filenames to represent the internal font name (truncating Condensed to Cond)

**Comment by BoldMonday** [comment]
@vpicone All fonts have been named to their internal font names. Our tools always take that info directly from the fonts themselves.

But fonts can contain **multiple** naming schemes -- there is naming for style mapping (which has a limit in length) and there is preferred naming. Please see https://docs.microsoft.com/en-us/typography/opentype/spec/name#name-ids for a full description of the name table in a font and in particular ID 1/2 and ID 16/17.

In case of Plex Sans Condensed the naming for style mapping uses “Cond” whereas the preferred naming uses “Condensed”:
```
Stylemap Family Name     | Stylemap Style Name
-------------------------+--------------------
IBM Plex Sans Cond Thin  | Regular
IBM Plex Sans Cond Thin  | Italic
IBM Plex Sans Cond ExtLt | Regular
IBM Plex Sans Cond ExtLt | Italic
IBM Plex Sans Cond Light | Regular
IBM Plex Sans Cond Light | Italic
IBM Plex Sans Cond       | Regular
IBM Plex Sans Cond       | Italic
IBM Plex Sans Cond Text  | Regular
IBM Plex Sans Cond Text  | Italic
IBM Plex Sans Cond Medm  | Regular
IBM Plex Sans Cond Medm  | Italic
IBM Plex Sans Cond SmBld | Regular
IBM Plex Sans Cond SmBld | Italic
IBM Plex Sans Cond       | Bold
IBM Plex Sans Cond       | Bold Italic
```
```
Preferred Family Name   | Preferred Style Name
------------------------+---------------------
IBM Plex Sans Condensed | Thin
IBM Plex Sans Condensed | Thin Italic
IBM Plex Sans Condensed | ExtraLight
IBM Plex Sans Condensed | ExtraLight Italic
IBM Plex Sans Condensed | Light
IBM Plex Sans Condensed | Light Italic
IBM Plex Sans Condensed | Regular
IBM Plex Sans Condensed | Italic
IBM Plex Sans Condensed | Text
IBM Plex Sans Condensed | Text Italic
IBM Plex Sans Condensed | Medium
IBM Plex Sans Condensed | Medium Italic
IBM Plex Sans Condensed | SemiBold
IBM Plex Sans Condensed | SemiBold Italic
IBM Plex Sans Condensed | Bold
IBM Plex Sans Condensed | Bold Italic
```
Therefore there is no need to change the file names because they already reflect the preferred naming.
However what is necessary to change in the css files are the `local` parameters. These should contain the names I posted earlier in this thread. These names are derived from name table ID 4 and ID 6.

**Comment by vpicone** [comment]
@BoldMonday That tracks, yeah we generate the css with some hardcoded values. I'll have a fix up today.

**Comment by BoldMonday** [comment]
@vpicone Great. I can have a look at the PR.

**Comment by vpicone** [comment]
@BoldMonday just an update, it seems whoever wrote the sass generation script assumed the font name would be used for both the URL and the local file name so updating it results in this. This is a bit what I was afraid of and why I recommended renaming the files.

```
src: local('IBM Plex Sans Cond ExtraLight'),
    local('IBMPlexSansCond-ExtraLight'),
    url('#{$font-prefix}/IBM-Plex-Sans-Cond/fonts/complete/woff2/IBMPlexSansCond-ExtraLight.woff2') format('woff2'),
    url('#{$font-prefix}/IBM-Plex-Sans-Cond/fonts/complete/woff/IBMPlexSansCond-ExtraLight.woff') format('woff');
```

Would you prefer me fix the script to seperate the local()/file url logic or to rename the files to match what the script expects based on the family name change?

**Comment by vpicone** [comment]
You'll see the folders and files are using the truncated name as well after adjusting the family name to fix local bits.

**Comment by BoldMonday** [comment]
> @BoldMonday just an update, it seems whoever wrote the sass generation script assumed the font name would be used for both the URL and the local file name so updating it results in this. This is a bit what I was afraid of and why I recommended renaming the files.

@vpicone That was a really bad assumption by the person who wrote that script. ;)

If you look at the first table that I posted then the names needed for the `local` parameter for ExtraLight Italic must be `IBM Plex Sans Cond ExtLt Italic` and `IBMPlexSans-CondExtLtItalic`.
These names are hardcoded in the fonts and cannot always be automatically generated/derived by algorithms. We manually need to set these names sometimes ourselves when we generate fonts because of limits in length.

> Would you prefer me fix the script to seperate the local()/file url logic or to rename the files to match what the script expects based on the family name change?

I would really prefer to change the script to take input from a list of names that we supply. And let the filenames be based on the "preferred naming" as we already do.

**Comment by vpicone** [comment]
@BoldMonday roger that, I'll dig into it again today and tomorrow.

**Comment by vpicone** [comment]
@BoldMonday even before the change to Cond we were always formatting the PS Font name as `IBMPlexSansCondensed-Bold` (note that condensed comes before the hyphen). Are we sure that `Cond` is supposed to go _after_ the hyphen?

**Comment by BoldMonday** [comment]
@vpicone This has been a small oversight on our part. For some time there were discussions to change the overall family structure and define Condensed as a sub-family. Eventually it was decided not to do that but the PS Font Name was already set in our source files.

There are two solutions here: either change the naming in the font files (more work but also more consistent) or allow this as an exception. Technically it does not matter for how the fonts will function when installed.

**Comment by mjabbink** [comment]
I think changing the naming in the font files is the way to go. We want the consistency and same logic wherever possible.

**Comment by BoldMonday** [comment]
Ok. We’ll prioritise this and send updated files soon.

**Comment by vpicone** [comment]
@mjabbink @BoldMonday sounds good, I can release it as a patch with my PR.

**Comment by vpicone** [comment]
@mjabbink @BoldMonday any word on this?

**Comment by BoldMonday** [comment]
Not yet. Hopefully in the course of next week.

**Comment by mjabbink** [comment]
@BoldMonday This is a top priority for us Paul. It’s been 22 days.

**Comment by BoldMonday** [comment]
I agree this is a bit overdue. Expect an update tomorrow.

**Comment by BoldMonday** [comment]
@vpicone @mjabbink updated files are done. Please use the names below for the `local` parameter.
```
Full Font Name                       PS Font Name
---------------------------------------------------------------
IBM Plex Sans Cond Bold Italic   |   IBMPlexSansCond-BoldItalic
IBM Plex Sans Cond Bold          |   IBMPlexSansCond-Bold
IBM Plex Sans Cond ExtLt Italic  |   IBMPlexSansCond-ExtLtItalic
IBM Plex Sans Cond ExtLt         |   IBMPlexSansCond-ExtLt
IBM Plex Sans Cond Italic        |   IBMPlexSansCond-Italic
IBM Plex Sans Cond Light Italic  |   IBMPlexSansCond-LightItalic
IBM Plex Sans Cond Light         |   IBMPlexSansCond-Light
IBM Plex Sans Cond Medm Italic   |   IBMPlexSansCond-MedmItalic
IBM Plex Sans Cond Medm          |   IBMPlexSansCond-Medm
IBM Plex Sans Cond               |   IBMPlexSansCond
IBM Plex Sans Cond SmBld Italic  |   IBMPlexSansCond-SmBldItalic
IBM Plex Sans Cond SmBld         |   IBMPlexSansCond-SmBld
IBM Plex Sans Cond Text Italic   |   IBMPlexSansCond-TextItalic
IBM Plex Sans Cond Text          |   IBMPlexSansCond-Text
IBM Plex Sans Cond Thin Italic   |   IBMPlexSansCond-ThinIt
```

**Comment by vpicone** [comment]
@BoldMonday just noticed some of the font weights are abbreviated, would need to create a special case for condensed here as this too his gathered from the font file name.

I'm having trouble understanding the pattern, It seems like the abbreviations are a bit arbitrary (i.e. why truncate `IBMPlexSansCond-ThinIt` but not `IBMPlexSansCond-TextItalic`).

**Comment by BoldMonday** [comment]
Hi @vpicone, the abbreviations for certain weights are correct since there is a limit on the amount of characters these fields can have inside of the fonts. (And that last entry was cut off indeed. I edited the list to correct it.)

But I thought we already agreed that compiling these names from the file name is not a good idea?
Earlier I wrote:
> These names are hardcoded in the fonts and cannot always be automatically generated/derived by algorithms. We manually need to set these names sometimes ourselves when we generate fonts because of limits in length.

Shall I provide you a list with the names for the `local` parameters for all members of the Plex family?
